src/components/dock/dock.types.ts

```diff
@@ -0,0 +1,88 @@
+export interface DockItem {
+    /**
+     * A non-changing value to uniquely identify each item.
+     */
+    id: string;
+
+    /**
+     * Text to display for the item.
+     */
+    label: string;
+
+    /**
+     * Name of the icon to use.
+     */
+    icon: string;
+
+    /**
+     * Additional helper text for the button.
+     * Example usage can be a keyboard shortcut to activate the button.
+     */
+    helperLabel?: string;
+
+    /**
+     * Whether the button should indicate it is selected.
+     * The button should be selected when … (@Kiarokh: add description here please!)
+     */
+    selected?: boolean;
+
+    /**
+     * Used to specify a custom component to render as the dock button.
+     */
+    dockButton?: DockButton;
```

Comment:
This is for defining a custom component to render a button in the dock, instead of the default one we'd otherwise use. (I imagine it will be quite rare that anyone would use this…)

Reply:
Removed.